Pod/Classes/XExtensionItem.h

```diff
@@ -102,37 +96,23 @@ typedef UIImage *(^XExtensionItemThumbnailProvidingBlock)(CGSize suggestedSize);
  */
 @property (nonatomic, copy) NSDictionary *userInfo;
 
+- (instancetype)initWithURL:(NSURL *)URL;
+- (instancetype)initWithText:(NSString *)text;
+- (instancetype)initWithImage:(UIImage *)image;
+- (instancetype)initWithData:(NSData *)data ofType:(NSString *)typeIdentifier;
+
 /**
  Initialize a new instance with a placeholder item, whose type will be used by the activity controller to determine
- which activities and extensions are displayed, and an array of attachments that will actually be passed to the selected 
- activity/extension.
+ which activities and extensions are displayed.
  
  @param placeholderItem (Required) A placeholder item whose type will be used by the activity controller to determine
- which activities and extensions are displayed. System activities that do not accept `NSExtensionItem` input will 
- receive this placeholder instead.
- @param attachments     An array of media data associated with the extension item. These items must be of
- type `NSItemProvider` and will be passed to the selected activity/extension.
+ which activities and extensions are displayed.
+ @param typeIdentifier  (Optional) A uniform type identifier describing the type of content being shared.
+ @param itemBlock       (Optional) A block that can provide the item to be shared given the activity type.
  
  @return New item source instance.
  */
-- (instancetype)initWithPlaceholderItem:(id)placeholderItem
-                            attachments:(NSArray/*<NSItemProvider>*/ *)attachments NS_DESIGNATED_INITIALIZER;
-
-/**
- Initialize a new instance with placeholder data and a type identifier which will be used by the activity controller to 
- determine which activities and extensions are displayed, and an array of attachments that will actually be passed to 
- the selected activity/extension.
- 
- @param placeholderData    (Required) Placeholder data. System activities that do not accept `NSExtensionItem` input
- will receive this placeholder instead.
- @param dataTypeIdentifier (Required) The UTI for the placeholder data.
- @param attachments        An array of media data associated with the extension item. These items must be of
- 
- @return New item source instance.
- */
-- (instancetype)initWithPlaceholderData:(id)placeholderData
-                     dataTypeIdentifier:(NSString *)dataTypeIdentifier
-                            attachments:(NSArray/*<NSItemProvider>*/ *)attachments;
+- (instancetype)initWithPlaceholderItem:(id)placeholderItem typeIdentifier:(NSString *)typeIdentifier itemBlock:(XExtensionItemProvidingBlock)itemBlock NS_DESIGNATED_INITIALIZER;
```

Comment:
I’d break the arguments onto separate lines.

Is this type of feedback useful or should I just merge and open a new one to address stylistic issues?